samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -47,46 +47,46 @@
    * transformed {@link MessageStream}.
    *
    * @param mapFn the function to transform a message to another message
-   * @param <TM> the type of messages in the transformed {@link MessageStream}
+   * @param <OM> the type of messages in the transformed {@link MessageStream}
    * @return the transformed {@link MessageStream}
    */
-  <TM> MessageStream<TM> map(MapFunction<? super M, ? extends TM> mapFn);
+  <OM> MessageStream<OM> map(MapFunction<? super M, ? extends OM> mapFn);
 
   /**
    * Applies the provided 1:n function to transform a message in this {@link MessageStream}
    * to n messages in the transformed {@link MessageStream}
    *
    * @param flatMapFn the function to transform a message to zero or more messages
-   * @param <TM> the type of messages in the transformed {@link MessageStream}
+   * @param <OM> the type of messages in the transformed {@link MessageStream}
    * @return the transformed {@link MessageStream}
    */
-  <TM> MessageStream<TM> flatMap(FlatMapFunction<? super M, ? extends TM> flatMapFn);
+  <OM> MessageStream<OM> flatMap(FlatMapFunction<? super M, ? extends OM> flatMapFn);
 
   /**
    * Applies the provided function to messages in this {@link MessageStream} and returns the
-   * transformed {@link MessageStream}.
+   * filtered {@link MessageStream}.
    * <p>
    * The {@link Function} is a predicate which determines whether a message in this {@link MessageStream}
-   * should be retained in the transformed {@link MessageStream}.
+   * should be retained in the filtered {@link MessageStream}.
    *
-   * @param filterFn the predicate to filter messages from this {@link MessageStream}
+   * @param filterFn the predicate to filter messages from this {@link MessageStream}.
    * @return the transformed {@link MessageStream}
    */
   MessageStream<M> filter(FilterFunction<? super M> filterFn);
 
   /**
    * Allows sending messages in this {@link MessageStream} to an output system using the provided {@link SinkFunction}.
-   *
-   * NOTE: If the output is for a {@link org.apache.samza.system.SystemStream}, use
-   * {@link #sendTo(OutputStream)} instead. This transform should only be used to output to
-   * non-stream systems (e.g., an external database).
+   * <p>
+   * Offers more control over processing and sending messages than {@link #sendTo(OutputStream)} since
```

Comment:
This is a very helpful note, thanks. 

Might also be worth mentioning that this can also be used to send output on a channel that doesn't have a System impl